data/room/src/main/java/be/appwise/room/BaseRoomDao.kt

```diff
@@ -91,20 +96,38 @@ abstract class BaseRoomDao<T : BaseEntity>(private val tableName: String) {
 
     /**
      *  Deletes all entities that are not inside the provided list and insert / replace the entities
-     *  inside the list
+     *  inside the list with the exclusion of the provided [excludedIds].
      *
      *  @param entities the list of objects that needs to be inserted
+     *  @param excludedIds the ids to be excluded from the removal
      *  @return a list of the inserted ID's
      */
-    suspend fun insertManyDeleteOthers(entities: List<T>) : List<Long> {
-        val ids = entities.joinToString { "\'${it.id}\'" }
+    suspend fun insertManyDeleteOthers(entities: List<T>, excludedIds: List<Any>): List<Long> {
+        var ids = entities.joinToString { "\'${it.id}\'" }
+
+        if (ids.isNotEmpty() && excludedIds.isNotEmpty())
+            ids += ","
+
+        if (excludedIds.isNotEmpty())
+            ids += excludedIds.joinToString { "\'$it\'" }
 
         val query = SimpleSQLiteQuery("DELETE FROM $tableName WHERE $idColumnInfo NOT IN($ids);")
         deleteAllExceptIds(query)
 
         return insertMany(entities)
     }
 
+    /**
+     *  Deletes all entities that are not inside the provided list and insert / replace the entities
+     *  inside the list
+     *
+     *  @param entities the list of objects that needs to be inserted
+     *  @return a list of the inserted ID's
+     */
+    suspend fun insertManyDeleteOthers(entities: List<T>): List<Long> {
```

Comment:
Is this needed? I think you can add a default value to the existing method and it will be backwards compatible.
`insertManyDeleteOthers(entities: List<T>, excludedIds: List<Any> = emptyList())`